tests/shared_utils_test.py

```diff
@@ -56,3 +61,36 @@ def test_mask_by_extent():
     assert not mask_by_extent(
         np.vstack([points]), np.vstack(corners[:2])
     ), "Point should have been outside."
+
+
+def test_box_intersect():
```

Comment:
**suggestion (readability, tests):** better split the function in dedicated tests

- test_disjoint
- test_fully_inside
- test_corner_inside (4 cases in 2D, 8 in 3D)
- test_edge_inside (4 cases in 2D, 12 in 3D)
- test_face inside (6 cases in 3D)

also test the symmetric each time (`box_intersect(a,b)` and `box_intersect(b,a)`)

for test with multiple cases, use something like
```py
@pytest.mark.parameterize("box_a, box_b, expected_box", boxes_for_corner_intersect)
def test_corner_intersect(box_a, box_b, expected_box)
```

Reply:
Very nice. Will do.